### ISSUE TITLE
ruby/ruby@d2a0330

### ISSUE THREAD

**Issue by iliabylich** [whitequark/parser#465]
Upstream commit: https://github.com/ruby/ruby/commit/d2a03300d33ef6b24fd4492aa3917ce41b4512ae

@whitequark Could you confirm that we need those error messages please?

**Comment by whitequark** [comment]
We already handle them, in a slightly different way:
```
$ ./bin/ruby-parse -e $'\\foo'
(fragment:0):1:1: error: bare backslash only allowed before newline
(fragment:0):1: \foo
(fragment:0):1: ^
```

**Comment by iliabylich** [comment]
So, what's your verdict? Should we backport it?

**Comment by iliabylich** [comment]
No, I was asking about returning different error messages for < 25 and > 25.

**Comment by whitequark** [comment]
Yeah, new parser tokens is how MRI returns different error messages. In general, Parser doesn't try to replicate the errors of MRI exactly, the goal is to reject the same code MRI rejects and offer a legible explanation (because MRI used to have atrocious error messages and duplicating those exactly didn't make any sense).
